core/src/main/java/org/elasticsearch/common/geo/GeoPoint.java

```diff
@@ -168,4 +168,28 @@ public static GeoPoint fromGeohash(long geohashLong) {
     public static GeoPoint fromIndexLong(long indexLong) {
         return new GeoPoint().resetFromIndexHash(indexLong);
     }
+
+    /** Puts latitude in range of -90 to 90. */
+    public static double normalizeLatitude(double latitude) {
+        if (latitude >= -90 && latitude <= 90) {
+            return latitude; //common case, and avoids slight double precision shifting
+        }
```

Comment:
This doesn't seem worth doing to me: if normalization is our bottleneck then we have bigger issues?